scripts/scil_compute_NODDI_priors.py

```diff
@@ -0,0 +1,186 @@
+        '--md_min', dest='md_t',  type=float, default='0.003',
+        help='Minimal threshold of MD in mm2/s (voxels above that threshold '
+             'are considered for in the ventricles). [%(default)s]')
+    p.add_argument(
+        '--fa_min', type=float, default='0.7',
+        help='Minimal threshold of FA (voxels above that threshold are '
+             'considered in the single fiber mask). [%(default)s]')
+
+    p.add_argument(
+        '--roi_radius', default=20, type=int,
+        help='Radius of the region used to estimate the priors. The roi will '
+             'be a cube spanning from ROI_CENTER in each direction. '
+             '[%(default)s]')
+    p.add_argument(
+        '--roi_center', metavar='tuple(3)', nargs="+", type=int,
+        help='Center of the roi of size roi_radius used to estimate the '
+             'priors. [center of the 3D volume]')
+
+    p.add_argument(
+        '--output_1fiber', metavar='file',
+        help='Output path for the text file containing the single '
+             'fiber average value of AD. If not set, the file will not be '
+             'saved.')
+    p.add_argument(
+        '--mask_output_1fiber', metavar='file',
+        help='Output path for single fiber mask. If not set, the mask will '
+             'not be saved.')
+    p.add_argument(
+        '--output_ventricles', metavar='file',
+        help='Output path for the text file containing the ventricles average '
+             'value of MD. If not set, the file will not be saved.')
+    p.add_argument(
+        '--mask_output_ventricles', metavar='file',
+        help='Output path for the ventricule mask. If not set, the mask will '
+             'not be saved.')
+
+    add_overwrite_arg(p)
+    add_verbose_arg(p)
+
+    return p
+
+
+def load(path):
+    img = nib.load(path)
+    return img.get_data(), img.affine, img.header.get_zooms()[:3]
+
+
+def save(data, affine, output):
+    img = nib.Nifti1Image(np.array(data, 'float32'),  affine)
+    nib.save(img, output)
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+
+    assert_inputs_exist(parser, [args.ad, args.fa, args.md])
+    assert_outputs_exist(parser, args, [],
+                         [args.mask_output_1fiber,
+                         args.mask_output_ventricles,
+                         args.output_ventricles,
+                         args.output_1fiber])
+
+    if args.verbose:
+        logging.basicConfig(level=logging.DEBUG)
+
+    fa, aff, hdr = load(args.fa)
+    md, _, _ = load(args.md)
+    ad, _, _ = load(args.ad)
+
+    mask_cc = np.zeros(fa.shape)
+    mask_vent = np.zeros(fa.shape)
+
+    # center
+    if args.roi_center is None:
+        ci, cj, ck = np.array(fa.shape[:3]) // 2
+    else:
+        if len(args.roi_center) != 3:
+            parser.error("roi_center needs to receive 3 values")
+        else:
+            ci, cj, ck = args.roi_center
+
+    w = args.roi_radius
+    roi_ad = ad[int(ci - w): int(ci + w),
```

Comment:
Are we happy with the wrap around behavior of the ROIs here ?

Reply:
I change a little bit the code so it's clearer.
What do you suggest ?

Reply:
I was wondering if it would be more consistent to clip the ranges to the borders of the image, since it does not make sense to have metrics that wraps around the image. If affects the mean and std extracted later.